packages/deployer/src/deploySolutionFromTemplate.ts

```diff
@@ -347,6 +335,33 @@ export function deploySolutionFromTemplate(
   });
 }
 
+/**
+ * Add source-id to items/groups typeKeywords
+ *
+ * @param template the array of solution data templates
+ * @internal
+ */
+export function _addSourceId(
+  templates: common.IItemTemplate[]
+): common.IItemTemplate[] {
+  return templates.map(
+    (template: any) => {
+      const sourceId: string = "source-" + template.itemId;
+      /* istanbul ignore else */
+      if (template.item) {
+        if (template.item!.typeKeywords) {
+          template.item!.typeKeywords!.push(sourceId);
+        } else /* istanbul ignore else */ if (common.getProp(template, "item.type") === "Group") {
```

Comment:
Is it necessary to test for the group type? This clause could simply handle any item/group that doesn't have `typeKeywords` for any reason.

Reply:
@MikeTschudi I wasn't sure if some item types could/would be expected to not have typeKeywords.

Reply:
Happy to update if you think it's best though

Reply:
The REST API documentation doesn't say what happens when there are no typeKeywords, so this struck me as a convenient way to survive the API not returning an empty list when there are no typeKeywords.